src/map/ai/states/magic_state.cpp

```diff
@@ -182,7 +182,7 @@ bool CMagicState::CanCastSpell(CBattleEntity* PTarget)
     {
         return false;
     }
-    if (distance(m_PEntity->loc.p, PTarget->loc.p) > 40)
```

Comment:
do not change this line - it is for the "too far away message" (as opposed to the "out of range" message, which is the next check).